// @aws-sdk/client-apigatewayv2/dist-cjs/models/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TooManyRequestsException = exports.NotFoundException = exports.ConflictException = exports.BadRequestException = exports.AccessDeniedException = void 0;
const ApiGatewayV2ServiceException_1 = require("./ApiGatewayV2ServiceException");
class AccessDeniedException extends ApiGatewayV2ServiceException_1.ApiGatewayV2ServiceException {
    name = "AccessDeniedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.Message = opts.Message;
    }
}
exports.AccessDeniedException = AccessDeniedException;
class BadRequestException extends ApiGatewayV2ServiceException_1.ApiGatewayV2ServiceException {
    name = "BadRequestException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "BadRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BadRequestException.prototype);
        this.Message = opts.Message;
    }
}
exports.BadRequestException = BadRequestException;
class ConflictException extends ApiGatewayV2ServiceException_1.ApiGatewayV2ServiceException {
    name = "ConflictException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
}
exports.ConflictException = ConflictException;
class NotFoundException extends ApiGatewayV2ServiceException_1.ApiGatewayV2ServiceException {
    name = "NotFoundException";
    $fault = "client";
    Message;
    ResourceType;
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotFoundException.prototype);
        this.Message = opts.Message;
        this.ResourceType = opts.ResourceType;
    }
}
exports.NotFoundException = NotFoundException;
class TooManyRequestsException extends ApiGatewayV2ServiceException_1.ApiGatewayV2ServiceException {
    name = "TooManyRequestsException";
    $fault = "client";
    LimitType;
    Message;
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.LimitType = opts.LimitType;
        this.Message = opts.Message;
    }
}
exports.TooManyRequestsException = TooManyRequestsException;
