// @aws-sdk/client-apigatewayv2/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var ApiGatewayV2ServiceException = require('./models/ApiGatewayV2ServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "apigateway",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class ApiGatewayV2Client extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultApiGatewayV2HttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class CreateApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateApi", {})
    .n("ApiGatewayV2Client", "CreateApiCommand")
    .sc(schemas_0.CreateApi$)
    .build() {
}

class CreateApiMappingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateApiMapping", {})
    .n("ApiGatewayV2Client", "CreateApiMappingCommand")
    .sc(schemas_0.CreateApiMapping$)
    .build() {
}

class CreateAuthorizerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateAuthorizer", {})
    .n("ApiGatewayV2Client", "CreateAuthorizerCommand")
    .sc(schemas_0.CreateAuthorizer$)
    .build() {
}

class CreateDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateDeployment", {})
    .n("ApiGatewayV2Client", "CreateDeploymentCommand")
    .sc(schemas_0.CreateDeployment$)
    .build() {
}

class CreateDomainNameCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateDomainName", {})
    .n("ApiGatewayV2Client", "CreateDomainNameCommand")
    .sc(schemas_0.CreateDomainName$)
    .build() {
}

class CreateIntegrationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateIntegration", {})
    .n("ApiGatewayV2Client", "CreateIntegrationCommand")
    .sc(schemas_0.CreateIntegration$)
    .build() {
}

class CreateIntegrationResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateIntegrationResponse", {})
    .n("ApiGatewayV2Client", "CreateIntegrationResponseCommand")
    .sc(schemas_0.CreateIntegrationResponse$)
    .build() {
}

class CreateModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateModel", {})
    .n("ApiGatewayV2Client", "CreateModelCommand")
    .sc(schemas_0.CreateModel$)
    .build() {
}

class CreatePortalCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreatePortal", {})
    .n("ApiGatewayV2Client", "CreatePortalCommand")
    .sc(schemas_0.CreatePortal$)
    .build() {
}

class CreatePortalProductCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreatePortalProduct", {})
    .n("ApiGatewayV2Client", "CreatePortalProductCommand")
    .sc(schemas_0.CreatePortalProduct$)
    .build() {
}

class CreateProductPageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateProductPage", {})
    .n("ApiGatewayV2Client", "CreateProductPageCommand")
    .sc(schemas_0.CreateProductPage$)
    .build() {
}

class CreateProductRestEndpointPageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateProductRestEndpointPage", {})
    .n("ApiGatewayV2Client", "CreateProductRestEndpointPageCommand")
    .sc(schemas_0.CreateProductRestEndpointPage$)
    .build() {
}

class CreateRouteCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateRoute", {})
    .n("ApiGatewayV2Client", "CreateRouteCommand")
    .sc(schemas_0.CreateRoute$)
    .build() {
}

class CreateRouteResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateRouteResponse", {})
    .n("ApiGatewayV2Client", "CreateRouteResponseCommand")
    .sc(schemas_0.CreateRouteResponse$)
    .build() {
}

class CreateRoutingRuleCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateRoutingRule", {})
    .n("ApiGatewayV2Client", "CreateRoutingRuleCommand")
    .sc(schemas_0.CreateRoutingRule$)
    .build() {
}

class CreateStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateStage", {})
    .n("ApiGatewayV2Client", "CreateStageCommand")
    .sc(schemas_0.CreateStage$)
    .build() {
}

class CreateVpcLinkCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "CreateVpcLink", {})
    .n("ApiGatewayV2Client", "CreateVpcLinkCommand")
    .sc(schemas_0.CreateVpcLink$)
    .build() {
}

class DeleteAccessLogSettingsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteAccessLogSettings", {})
    .n("ApiGatewayV2Client", "DeleteAccessLogSettingsCommand")
    .sc(schemas_0.DeleteAccessLogSettings$)
    .build() {
}

class DeleteApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteApi", {})
    .n("ApiGatewayV2Client", "DeleteApiCommand")
    .sc(schemas_0.DeleteApi$)
    .build() {
}

class DeleteApiMappingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteApiMapping", {})
    .n("ApiGatewayV2Client", "DeleteApiMappingCommand")
    .sc(schemas_0.DeleteApiMapping$)
    .build() {
}

class DeleteAuthorizerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteAuthorizer", {})
    .n("ApiGatewayV2Client", "DeleteAuthorizerCommand")
    .sc(schemas_0.DeleteAuthorizer$)
    .build() {
}

class DeleteCorsConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteCorsConfiguration", {})
    .n("ApiGatewayV2Client", "DeleteCorsConfigurationCommand")
    .sc(schemas_0.DeleteCorsConfiguration$)
    .build() {
}

class DeleteDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteDeployment", {})
    .n("ApiGatewayV2Client", "DeleteDeploymentCommand")
    .sc(schemas_0.DeleteDeployment$)
    .build() {
}

class DeleteDomainNameCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteDomainName", {})
    .n("ApiGatewayV2Client", "DeleteDomainNameCommand")
    .sc(schemas_0.DeleteDomainName$)
    .build() {
}

class DeleteIntegrationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteIntegration", {})
    .n("ApiGatewayV2Client", "DeleteIntegrationCommand")
    .sc(schemas_0.DeleteIntegration$)
    .build() {
}

class DeleteIntegrationResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteIntegrationResponse", {})
    .n("ApiGatewayV2Client", "DeleteIntegrationResponseCommand")
    .sc(schemas_0.DeleteIntegrationResponse$)
    .build() {
}

class DeleteModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteModel", {})
    .n("ApiGatewayV2Client", "DeleteModelCommand")
    .sc(schemas_0.DeleteModel$)
    .build() {
}

class DeletePortalCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeletePortal", {})
    .n("ApiGatewayV2Client", "DeletePortalCommand")
    .sc(schemas_0.DeletePortal$)
    .build() {
}

class DeletePortalProductCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeletePortalProduct", {})
    .n("ApiGatewayV2Client", "DeletePortalProductCommand")
    .sc(schemas_0.DeletePortalProduct$)
    .build() {
}

class DeletePortalProductSharingPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeletePortalProductSharingPolicy", {})
    .n("ApiGatewayV2Client", "DeletePortalProductSharingPolicyCommand")
    .sc(schemas_0.DeletePortalProductSharingPolicy$)
    .build() {
}

class DeleteProductPageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteProductPage", {})
    .n("ApiGatewayV2Client", "DeleteProductPageCommand")
    .sc(schemas_0.DeleteProductPage$)
    .build() {
}

class DeleteProductRestEndpointPageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteProductRestEndpointPage", {})
    .n("ApiGatewayV2Client", "DeleteProductRestEndpointPageCommand")
    .sc(schemas_0.DeleteProductRestEndpointPage$)
    .build() {
}

class DeleteRouteCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteRoute", {})
    .n("ApiGatewayV2Client", "DeleteRouteCommand")
    .sc(schemas_0.DeleteRoute$)
    .build() {
}

class DeleteRouteRequestParameterCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteRouteRequestParameter", {})
    .n("ApiGatewayV2Client", "DeleteRouteRequestParameterCommand")
    .sc(schemas_0.DeleteRouteRequestParameter$)
    .build() {
}

class DeleteRouteResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteRouteResponse", {})
    .n("ApiGatewayV2Client", "DeleteRouteResponseCommand")
    .sc(schemas_0.DeleteRouteResponse$)
    .build() {
}

class DeleteRouteSettingsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteRouteSettings", {})
    .n("ApiGatewayV2Client", "DeleteRouteSettingsCommand")
    .sc(schemas_0.DeleteRouteSettings$)
    .build() {
}

class DeleteRoutingRuleCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteRoutingRule", {})
    .n("ApiGatewayV2Client", "DeleteRoutingRuleCommand")
    .sc(schemas_0.DeleteRoutingRule$)
    .build() {
}

class DeleteStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteStage", {})
    .n("ApiGatewayV2Client", "DeleteStageCommand")
    .sc(schemas_0.DeleteStage$)
    .build() {
}

class DeleteVpcLinkCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DeleteVpcLink", {})
    .n("ApiGatewayV2Client", "DeleteVpcLinkCommand")
    .sc(schemas_0.DeleteVpcLink$)
    .build() {
}

class DisablePortalCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "DisablePortal", {})
    .n("ApiGatewayV2Client", "DisablePortalCommand")
    .sc(schemas_0.DisablePortal$)
    .build() {
}

class ExportApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "ExportApi", {})
    .n("ApiGatewayV2Client", "ExportApiCommand")
    .sc(schemas_0.ExportApi$)
    .build() {
}

class GetApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetApi", {})
    .n("ApiGatewayV2Client", "GetApiCommand")
    .sc(schemas_0.GetApi$)
    .build() {
}

class GetApiMappingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetApiMapping", {})
    .n("ApiGatewayV2Client", "GetApiMappingCommand")
    .sc(schemas_0.GetApiMapping$)
    .build() {
}

class GetApiMappingsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetApiMappings", {})
    .n("ApiGatewayV2Client", "GetApiMappingsCommand")
    .sc(schemas_0.GetApiMappings$)
    .build() {
}

class GetApisCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetApis", {})
    .n("ApiGatewayV2Client", "GetApisCommand")
    .sc(schemas_0.GetApis$)
    .build() {
}

class GetAuthorizerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetAuthorizer", {})
    .n("ApiGatewayV2Client", "GetAuthorizerCommand")
    .sc(schemas_0.GetAuthorizer$)
    .build() {
}

class GetAuthorizersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetAuthorizers", {})
    .n("ApiGatewayV2Client", "GetAuthorizersCommand")
    .sc(schemas_0.GetAuthorizers$)
    .build() {
}

class GetDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetDeployment", {})
    .n("ApiGatewayV2Client", "GetDeploymentCommand")
    .sc(schemas_0.GetDeployment$)
    .build() {
}

class GetDeploymentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetDeployments", {})
    .n("ApiGatewayV2Client", "GetDeploymentsCommand")
    .sc(schemas_0.GetDeployments$)
    .build() {
}

class GetDomainNameCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetDomainName", {})
    .n("ApiGatewayV2Client", "GetDomainNameCommand")
    .sc(schemas_0.GetDomainName$)
    .build() {
}

class GetDomainNamesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetDomainNames", {})
    .n("ApiGatewayV2Client", "GetDomainNamesCommand")
    .sc(schemas_0.GetDomainNames$)
    .build() {
}

class GetIntegrationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetIntegration", {})
    .n("ApiGatewayV2Client", "GetIntegrationCommand")
    .sc(schemas_0.GetIntegration$)
    .build() {
}

class GetIntegrationResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetIntegrationResponse", {})
    .n("ApiGatewayV2Client", "GetIntegrationResponseCommand")
    .sc(schemas_0.GetIntegrationResponse$)
    .build() {
}

class GetIntegrationResponsesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetIntegrationResponses", {})
    .n("ApiGatewayV2Client", "GetIntegrationResponsesCommand")
    .sc(schemas_0.GetIntegrationResponses$)
    .build() {
}

class GetIntegrationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetIntegrations", {})
    .n("ApiGatewayV2Client", "GetIntegrationsCommand")
    .sc(schemas_0.GetIntegrations$)
    .build() {
}

class GetModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetModel", {})
    .n("ApiGatewayV2Client", "GetModelCommand")
    .sc(schemas_0.GetModel$)
    .build() {
}

class GetModelsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetModels", {})
    .n("ApiGatewayV2Client", "GetModelsCommand")
    .sc(schemas_0.GetModels$)
    .build() {
}

class GetModelTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetModelTemplate", {})
    .n("ApiGatewayV2Client", "GetModelTemplateCommand")
    .sc(schemas_0.GetModelTemplate$)
    .build() {
}

class GetPortalCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetPortal", {})
    .n("ApiGatewayV2Client", "GetPortalCommand")
    .sc(schemas_0.GetPortal$)
    .build() {
}

class GetPortalProductCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetPortalProduct", {})
    .n("ApiGatewayV2Client", "GetPortalProductCommand")
    .sc(schemas_0.GetPortalProduct$)
    .build() {
}

class GetPortalProductSharingPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetPortalProductSharingPolicy", {})
    .n("ApiGatewayV2Client", "GetPortalProductSharingPolicyCommand")
    .sc(schemas_0.GetPortalProductSharingPolicy$)
    .build() {
}

class GetProductPageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetProductPage", {})
    .n("ApiGatewayV2Client", "GetProductPageCommand")
    .sc(schemas_0.GetProductPage$)
    .build() {
}

class GetProductRestEndpointPageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetProductRestEndpointPage", {})
    .n("ApiGatewayV2Client", "GetProductRestEndpointPageCommand")
    .sc(schemas_0.GetProductRestEndpointPage$)
    .build() {
}

class GetRouteCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetRoute", {})
    .n("ApiGatewayV2Client", "GetRouteCommand")
    .sc(schemas_0.GetRoute$)
    .build() {
}

class GetRouteResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetRouteResponse", {})
    .n("ApiGatewayV2Client", "GetRouteResponseCommand")
    .sc(schemas_0.GetRouteResponse$)
    .build() {
}

class GetRouteResponsesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetRouteResponses", {})
    .n("ApiGatewayV2Client", "GetRouteResponsesCommand")
    .sc(schemas_0.GetRouteResponses$)
    .build() {
}

class GetRoutesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetRoutes", {})
    .n("ApiGatewayV2Client", "GetRoutesCommand")
    .sc(schemas_0.GetRoutes$)
    .build() {
}

class GetRoutingRuleCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetRoutingRule", {})
    .n("ApiGatewayV2Client", "GetRoutingRuleCommand")
    .sc(schemas_0.GetRoutingRule$)
    .build() {
}

class GetStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetStage", {})
    .n("ApiGatewayV2Client", "GetStageCommand")
    .sc(schemas_0.GetStage$)
    .build() {
}

class GetStagesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetStages", {})
    .n("ApiGatewayV2Client", "GetStagesCommand")
    .sc(schemas_0.GetStages$)
    .build() {
}

class GetTagsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetTags", {})
    .n("ApiGatewayV2Client", "GetTagsCommand")
    .sc(schemas_0.GetTags$)
    .build() {
}

class GetVpcLinkCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetVpcLink", {})
    .n("ApiGatewayV2Client", "GetVpcLinkCommand")
    .sc(schemas_0.GetVpcLink$)
    .build() {
}

class GetVpcLinksCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "GetVpcLinks", {})
    .n("ApiGatewayV2Client", "GetVpcLinksCommand")
    .sc(schemas_0.GetVpcLinks$)
    .build() {
}

class ImportApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "ImportApi", {})
    .n("ApiGatewayV2Client", "ImportApiCommand")
    .sc(schemas_0.ImportApi$)
    .build() {
}

class ListPortalProductsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "ListPortalProducts", {})
    .n("ApiGatewayV2Client", "ListPortalProductsCommand")
    .sc(schemas_0.ListPortalProducts$)
    .build() {
}

class ListPortalsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "ListPortals", {})
    .n("ApiGatewayV2Client", "ListPortalsCommand")
    .sc(schemas_0.ListPortals$)
    .build() {
}

class ListProductPagesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "ListProductPages", {})
    .n("ApiGatewayV2Client", "ListProductPagesCommand")
    .sc(schemas_0.ListProductPages$)
    .build() {
}

class ListProductRestEndpointPagesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "ListProductRestEndpointPages", {})
    .n("ApiGatewayV2Client", "ListProductRestEndpointPagesCommand")
    .sc(schemas_0.ListProductRestEndpointPages$)
    .build() {
}

class ListRoutingRulesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "ListRoutingRules", {})
    .n("ApiGatewayV2Client", "ListRoutingRulesCommand")
    .sc(schemas_0.ListRoutingRules$)
    .build() {
}

class PreviewPortalCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "PreviewPortal", {})
    .n("ApiGatewayV2Client", "PreviewPortalCommand")
    .sc(schemas_0.PreviewPortal$)
    .build() {
}

class PublishPortalCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "PublishPortal", {})
    .n("ApiGatewayV2Client", "PublishPortalCommand")
    .sc(schemas_0.PublishPortal$)
    .build() {
}

class PutPortalProductSharingPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "PutPortalProductSharingPolicy", {})
    .n("ApiGatewayV2Client", "PutPortalProductSharingPolicyCommand")
    .sc(schemas_0.PutPortalProductSharingPolicy$)
    .build() {
}

class PutRoutingRuleCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "PutRoutingRule", {})
    .n("ApiGatewayV2Client", "PutRoutingRuleCommand")
    .sc(schemas_0.PutRoutingRule$)
    .build() {
}

class ReimportApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "ReimportApi", {})
    .n("ApiGatewayV2Client", "ReimportApiCommand")
    .sc(schemas_0.ReimportApi$)
    .build() {
}

class ResetAuthorizersCacheCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "ResetAuthorizersCache", {})
    .n("ApiGatewayV2Client", "ResetAuthorizersCacheCommand")
    .sc(schemas_0.ResetAuthorizersCache$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "TagResource", {})
    .n("ApiGatewayV2Client", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UntagResource", {})
    .n("ApiGatewayV2Client", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateApi", {})
    .n("ApiGatewayV2Client", "UpdateApiCommand")
    .sc(schemas_0.UpdateApi$)
    .build() {
}

class UpdateApiMappingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateApiMapping", {})
    .n("ApiGatewayV2Client", "UpdateApiMappingCommand")
    .sc(schemas_0.UpdateApiMapping$)
    .build() {
}

class UpdateAuthorizerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateAuthorizer", {})
    .n("ApiGatewayV2Client", "UpdateAuthorizerCommand")
    .sc(schemas_0.UpdateAuthorizer$)
    .build() {
}

class UpdateDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateDeployment", {})
    .n("ApiGatewayV2Client", "UpdateDeploymentCommand")
    .sc(schemas_0.UpdateDeployment$)
    .build() {
}

class UpdateDomainNameCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateDomainName", {})
    .n("ApiGatewayV2Client", "UpdateDomainNameCommand")
    .sc(schemas_0.UpdateDomainName$)
    .build() {
}

class UpdateIntegrationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateIntegration", {})
    .n("ApiGatewayV2Client", "UpdateIntegrationCommand")
    .sc(schemas_0.UpdateIntegration$)
    .build() {
}

class UpdateIntegrationResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateIntegrationResponse", {})
    .n("ApiGatewayV2Client", "UpdateIntegrationResponseCommand")
    .sc(schemas_0.UpdateIntegrationResponse$)
    .build() {
}

class UpdateModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateModel", {})
    .n("ApiGatewayV2Client", "UpdateModelCommand")
    .sc(schemas_0.UpdateModel$)
    .build() {
}

class UpdatePortalCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdatePortal", {})
    .n("ApiGatewayV2Client", "UpdatePortalCommand")
    .sc(schemas_0.UpdatePortal$)
    .build() {
}

class UpdatePortalProductCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdatePortalProduct", {})
    .n("ApiGatewayV2Client", "UpdatePortalProductCommand")
    .sc(schemas_0.UpdatePortalProduct$)
    .build() {
}

class UpdateProductPageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateProductPage", {})
    .n("ApiGatewayV2Client", "UpdateProductPageCommand")
    .sc(schemas_0.UpdateProductPage$)
    .build() {
}

class UpdateProductRestEndpointPageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateProductRestEndpointPage", {})
    .n("ApiGatewayV2Client", "UpdateProductRestEndpointPageCommand")
    .sc(schemas_0.UpdateProductRestEndpointPage$)
    .build() {
}

class UpdateRouteCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateRoute", {})
    .n("ApiGatewayV2Client", "UpdateRouteCommand")
    .sc(schemas_0.UpdateRoute$)
    .build() {
}

class UpdateRouteResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateRouteResponse", {})
    .n("ApiGatewayV2Client", "UpdateRouteResponseCommand")
    .sc(schemas_0.UpdateRouteResponse$)
    .build() {
}

class UpdateStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateStage", {})
    .n("ApiGatewayV2Client", "UpdateStageCommand")
    .sc(schemas_0.UpdateStage$)
    .build() {
}

class UpdateVpcLinkCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ApiGatewayV2", "UpdateVpcLink", {})
    .n("ApiGatewayV2Client", "UpdateVpcLinkCommand")
    .sc(schemas_0.UpdateVpcLink$)
    .build() {
}

const paginateListRoutingRules = core.createPaginator(ApiGatewayV2Client, ListRoutingRulesCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    CreateApiCommand,
    CreateApiMappingCommand,
    CreateAuthorizerCommand,
    CreateDeploymentCommand,
    CreateDomainNameCommand,
    CreateIntegrationCommand,
    CreateIntegrationResponseCommand,
    CreateModelCommand,
    CreatePortalCommand,
    CreatePortalProductCommand,
    CreateProductPageCommand,
    CreateProductRestEndpointPageCommand,
    CreateRouteCommand,
    CreateRouteResponseCommand,
    CreateRoutingRuleCommand,
    CreateStageCommand,
    CreateVpcLinkCommand,
    DeleteAccessLogSettingsCommand,
    DeleteApiCommand,
    DeleteApiMappingCommand,
    DeleteAuthorizerCommand,
    DeleteCorsConfigurationCommand,
    DeleteDeploymentCommand,
    DeleteDomainNameCommand,
    DeleteIntegrationCommand,
    DeleteIntegrationResponseCommand,
    DeleteModelCommand,
    DeletePortalCommand,
    DeletePortalProductCommand,
    DeletePortalProductSharingPolicyCommand,
    DeleteProductPageCommand,
    DeleteProductRestEndpointPageCommand,
    DeleteRouteCommand,
    DeleteRouteRequestParameterCommand,
    DeleteRouteResponseCommand,
    DeleteRouteSettingsCommand,
    DeleteRoutingRuleCommand,
    DeleteStageCommand,
    DeleteVpcLinkCommand,
    DisablePortalCommand,
    ExportApiCommand,
    GetApiCommand,
    GetApiMappingCommand,
    GetApiMappingsCommand,
    GetApisCommand,
    GetAuthorizerCommand,
    GetAuthorizersCommand,
    GetDeploymentCommand,
    GetDeploymentsCommand,
    GetDomainNameCommand,
    GetDomainNamesCommand,
    GetIntegrationCommand,
    GetIntegrationResponseCommand,
    GetIntegrationResponsesCommand,
    GetIntegrationsCommand,
    GetModelCommand,
    GetModelsCommand,
    GetModelTemplateCommand,
    GetPortalCommand,
    GetPortalProductCommand,
    GetPortalProductSharingPolicyCommand,
    GetProductPageCommand,
    GetProductRestEndpointPageCommand,
    GetRouteCommand,
    GetRouteResponseCommand,
    GetRouteResponsesCommand,
    GetRoutesCommand,
    GetRoutingRuleCommand,
    GetStageCommand,
    GetStagesCommand,
    GetTagsCommand,
    GetVpcLinkCommand,
    GetVpcLinksCommand,
    ImportApiCommand,
    ListPortalProductsCommand,
    ListPortalsCommand,
    ListProductPagesCommand,
    ListProductRestEndpointPagesCommand,
    ListRoutingRulesCommand,
    PreviewPortalCommand,
    PublishPortalCommand,
    PutPortalProductSharingPolicyCommand,
    PutRoutingRuleCommand,
    ReimportApiCommand,
    ResetAuthorizersCacheCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateApiCommand,
    UpdateApiMappingCommand,
    UpdateAuthorizerCommand,
    UpdateDeploymentCommand,
    UpdateDomainNameCommand,
    UpdateIntegrationCommand,
    UpdateIntegrationResponseCommand,
    UpdateModelCommand,
    UpdatePortalCommand,
    UpdatePortalProductCommand,
    UpdateProductPageCommand,
    UpdateProductRestEndpointPageCommand,
    UpdateRouteCommand,
    UpdateRouteResponseCommand,
    UpdateStageCommand,
    UpdateVpcLinkCommand,
};
const paginators = {
    paginateListRoutingRules,
};
class ApiGatewayV2 extends ApiGatewayV2Client {
}
smithyClient.createAggregatedClient(commands, ApiGatewayV2, { paginators });

const IpAddressType = {
    dualstack: "dualstack",
    ipv4: "ipv4",
};
const ProtocolType = {
    HTTP: "HTTP",
    WEBSOCKET: "WEBSOCKET",
};
const AuthorizerType = {
    JWT: "JWT",
    REQUEST: "REQUEST",
};
const DeploymentStatus = {
    DEPLOYED: "DEPLOYED",
    FAILED: "FAILED",
    PENDING: "PENDING",
};
const DomainNameStatus = {
    AVAILABLE: "AVAILABLE",
    PENDING_CERTIFICATE_REIMPORT: "PENDING_CERTIFICATE_REIMPORT",
    PENDING_OWNERSHIP_VERIFICATION: "PENDING_OWNERSHIP_VERIFICATION",
    UPDATING: "UPDATING",
};
const EndpointType = {
    EDGE: "EDGE",
    REGIONAL: "REGIONAL",
};
const SecurityPolicy = {
    TLS_1_0: "TLS_1_0",
    TLS_1_2: "TLS_1_2",
};
const RoutingMode = {
    API_MAPPING_ONLY: "API_MAPPING_ONLY",
    ROUTING_RULE_ONLY: "ROUTING_RULE_ONLY",
    ROUTING_RULE_THEN_API_MAPPING: "ROUTING_RULE_THEN_API_MAPPING",
};
const ConnectionType = {
    INTERNET: "INTERNET",
    VPC_LINK: "VPC_LINK",
};
const ContentHandlingStrategy = {
    CONVERT_TO_BINARY: "CONVERT_TO_BINARY",
    CONVERT_TO_TEXT: "CONVERT_TO_TEXT",
};
const IntegrationType = {
    AWS: "AWS",
    AWS_PROXY: "AWS_PROXY",
    HTTP: "HTTP",
    HTTP_PROXY: "HTTP_PROXY",
    MOCK: "MOCK",
};
const PassthroughBehavior = {
    NEVER: "NEVER",
    WHEN_NO_MATCH: "WHEN_NO_MATCH",
    WHEN_NO_TEMPLATES: "WHEN_NO_TEMPLATES",
};
const PreviewStatus = {
    PREVIEW_FAILED: "PREVIEW_FAILED",
    PREVIEW_IN_PROGRESS: "PREVIEW_IN_PROGRESS",
    PREVIEW_READY: "PREVIEW_READY",
};
const PublishStatus = {
    DISABLED: "DISABLED",
    PUBLISHED: "PUBLISHED",
    PUBLISH_FAILED: "PUBLISH_FAILED",
    PUBLISH_IN_PROGRESS: "PUBLISH_IN_PROGRESS",
};
const Status = {
    AVAILABLE: "AVAILABLE",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
};
const TryItState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const AuthorizationType = {
    AWS_IAM: "AWS_IAM",
    CUSTOM: "CUSTOM",
    JWT: "JWT",
    NONE: "NONE",
};
const LoggingLevel = {
    ERROR: "ERROR",
    INFO: "INFO",
    OFF: "OFF",
};
const VpcLinkStatus = {
    AVAILABLE: "AVAILABLE",
    DELETING: "DELETING",
    FAILED: "FAILED",
    INACTIVE: "INACTIVE",
    PENDING: "PENDING",
};
const VpcLinkVersion = {
    V2: "V2",
};

exports.$Command = smithyClient.Command;
exports.__Client = smithyClient.Client;
exports.ApiGatewayV2ServiceException = ApiGatewayV2ServiceException.ApiGatewayV2ServiceException;
exports.ApiGatewayV2 = ApiGatewayV2;
exports.ApiGatewayV2Client = ApiGatewayV2Client;
exports.AuthorizationType = AuthorizationType;
exports.AuthorizerType = AuthorizerType;
exports.ConnectionType = ConnectionType;
exports.ContentHandlingStrategy = ContentHandlingStrategy;
exports.CreateApiCommand = CreateApiCommand;
exports.CreateApiMappingCommand = CreateApiMappingCommand;
exports.CreateAuthorizerCommand = CreateAuthorizerCommand;
exports.CreateDeploymentCommand = CreateDeploymentCommand;
exports.CreateDomainNameCommand = CreateDomainNameCommand;
exports.CreateIntegrationCommand = CreateIntegrationCommand;
exports.CreateIntegrationResponseCommand = CreateIntegrationResponseCommand;
exports.CreateModelCommand = CreateModelCommand;
exports.CreatePortalCommand = CreatePortalCommand;
exports.CreatePortalProductCommand = CreatePortalProductCommand;
exports.CreateProductPageCommand = CreateProductPageCommand;
exports.CreateProductRestEndpointPageCommand = CreateProductRestEndpointPageCommand;
exports.CreateRouteCommand = CreateRouteCommand;
exports.CreateRouteResponseCommand = CreateRouteResponseCommand;
exports.CreateRoutingRuleCommand = CreateRoutingRuleCommand;
exports.CreateStageCommand = CreateStageCommand;
exports.CreateVpcLinkCommand = CreateVpcLinkCommand;
exports.DeleteAccessLogSettingsCommand = DeleteAccessLogSettingsCommand;
exports.DeleteApiCommand = DeleteApiCommand;
exports.DeleteApiMappingCommand = DeleteApiMappingCommand;
exports.DeleteAuthorizerCommand = DeleteAuthorizerCommand;
exports.DeleteCorsConfigurationCommand = DeleteCorsConfigurationCommand;
exports.DeleteDeploymentCommand = DeleteDeploymentCommand;
exports.DeleteDomainNameCommand = DeleteDomainNameCommand;
exports.DeleteIntegrationCommand = DeleteIntegrationCommand;
exports.DeleteIntegrationResponseCommand = DeleteIntegrationResponseCommand;
exports.DeleteModelCommand = DeleteModelCommand;
exports.DeletePortalCommand = DeletePortalCommand;
exports.DeletePortalProductCommand = DeletePortalProductCommand;
exports.DeletePortalProductSharingPolicyCommand = DeletePortalProductSharingPolicyCommand;
exports.DeleteProductPageCommand = DeleteProductPageCommand;
exports.DeleteProductRestEndpointPageCommand = DeleteProductRestEndpointPageCommand;
exports.DeleteRouteCommand = DeleteRouteCommand;
exports.DeleteRouteRequestParameterCommand = DeleteRouteRequestParameterCommand;
exports.DeleteRouteResponseCommand = DeleteRouteResponseCommand;
exports.DeleteRouteSettingsCommand = DeleteRouteSettingsCommand;
exports.DeleteRoutingRuleCommand = DeleteRoutingRuleCommand;
exports.DeleteStageCommand = DeleteStageCommand;
exports.DeleteVpcLinkCommand = DeleteVpcLinkCommand;
exports.DeploymentStatus = DeploymentStatus;
exports.DisablePortalCommand = DisablePortalCommand;
exports.DomainNameStatus = DomainNameStatus;
exports.EndpointType = EndpointType;
exports.ExportApiCommand = ExportApiCommand;
exports.GetApiCommand = GetApiCommand;
exports.GetApiMappingCommand = GetApiMappingCommand;
exports.GetApiMappingsCommand = GetApiMappingsCommand;
exports.GetApisCommand = GetApisCommand;
exports.GetAuthorizerCommand = GetAuthorizerCommand;
exports.GetAuthorizersCommand = GetAuthorizersCommand;
exports.GetDeploymentCommand = GetDeploymentCommand;
exports.GetDeploymentsCommand = GetDeploymentsCommand;
exports.GetDomainNameCommand = GetDomainNameCommand;
exports.GetDomainNamesCommand = GetDomainNamesCommand;
exports.GetIntegrationCommand = GetIntegrationCommand;
exports.GetIntegrationResponseCommand = GetIntegrationResponseCommand;
exports.GetIntegrationResponsesCommand = GetIntegrationResponsesCommand;
exports.GetIntegrationsCommand = GetIntegrationsCommand;
exports.GetModelCommand = GetModelCommand;
exports.GetModelTemplateCommand = GetModelTemplateCommand;
exports.GetModelsCommand = GetModelsCommand;
exports.GetPortalCommand = GetPortalCommand;
exports.GetPortalProductCommand = GetPortalProductCommand;
exports.GetPortalProductSharingPolicyCommand = GetPortalProductSharingPolicyCommand;
exports.GetProductPageCommand = GetProductPageCommand;
exports.GetProductRestEndpointPageCommand = GetProductRestEndpointPageCommand;
exports.GetRouteCommand = GetRouteCommand;
exports.GetRouteResponseCommand = GetRouteResponseCommand;
exports.GetRouteResponsesCommand = GetRouteResponsesCommand;
exports.GetRoutesCommand = GetRoutesCommand;
exports.GetRoutingRuleCommand = GetRoutingRuleCommand;
exports.GetStageCommand = GetStageCommand;
exports.GetStagesCommand = GetStagesCommand;
exports.GetTagsCommand = GetTagsCommand;
exports.GetVpcLinkCommand = GetVpcLinkCommand;
exports.GetVpcLinksCommand = GetVpcLinksCommand;
exports.ImportApiCommand = ImportApiCommand;
exports.IntegrationType = IntegrationType;
exports.IpAddressType = IpAddressType;
exports.ListPortalProductsCommand = ListPortalProductsCommand;
exports.ListPortalsCommand = ListPortalsCommand;
exports.ListProductPagesCommand = ListProductPagesCommand;
exports.ListProductRestEndpointPagesCommand = ListProductRestEndpointPagesCommand;
exports.ListRoutingRulesCommand = ListRoutingRulesCommand;
exports.LoggingLevel = LoggingLevel;
exports.PassthroughBehavior = PassthroughBehavior;
exports.PreviewPortalCommand = PreviewPortalCommand;
exports.PreviewStatus = PreviewStatus;
exports.ProtocolType = ProtocolType;
exports.PublishPortalCommand = PublishPortalCommand;
exports.PublishStatus = PublishStatus;
exports.PutPortalProductSharingPolicyCommand = PutPortalProductSharingPolicyCommand;
exports.PutRoutingRuleCommand = PutRoutingRuleCommand;
exports.ReimportApiCommand = ReimportApiCommand;
exports.ResetAuthorizersCacheCommand = ResetAuthorizersCacheCommand;
exports.RoutingMode = RoutingMode;
exports.SecurityPolicy = SecurityPolicy;
exports.Status = Status;
exports.TagResourceCommand = TagResourceCommand;
exports.TryItState = TryItState;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateApiCommand = UpdateApiCommand;
exports.UpdateApiMappingCommand = UpdateApiMappingCommand;
exports.UpdateAuthorizerCommand = UpdateAuthorizerCommand;
exports.UpdateDeploymentCommand = UpdateDeploymentCommand;
exports.UpdateDomainNameCommand = UpdateDomainNameCommand;
exports.UpdateIntegrationCommand = UpdateIntegrationCommand;
exports.UpdateIntegrationResponseCommand = UpdateIntegrationResponseCommand;
exports.UpdateModelCommand = UpdateModelCommand;
exports.UpdatePortalCommand = UpdatePortalCommand;
exports.UpdatePortalProductCommand = UpdatePortalProductCommand;
exports.UpdateProductPageCommand = UpdateProductPageCommand;
exports.UpdateProductRestEndpointPageCommand = UpdateProductRestEndpointPageCommand;
exports.UpdateRouteCommand = UpdateRouteCommand;
exports.UpdateRouteResponseCommand = UpdateRouteResponseCommand;
exports.UpdateStageCommand = UpdateStageCommand;
exports.UpdateVpcLinkCommand = UpdateVpcLinkCommand;
exports.VpcLinkStatus = VpcLinkStatus;
exports.VpcLinkVersion = VpcLinkVersion;
exports.paginateListRoutingRules = paginateListRoutingRules;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
